qoqo/Half_Adder_Example.ipynb

```diff
@@ -0,0 +1,324 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example: Half Adder\n",
+    "In this example we write a quantum algorithm to perform an add operation between two qubits and store the result in two qubits that act as output register.\n",
+    "\n",
+    "The truth table of the circuit is\n",
+    "| in q1 | in q2 | out q4 | out q3 | out  (integer) |\n",
+    "|-------|-------|--------|--------|----------------|\n",
+    "| 0     | 0     | 0      | 0      | 0              |\n",
+    "| 0     | 1     | 0      | 1      | 1              |\n",
+    "| 1     | 0     | 0      | 1      | 1              |\n",
+    "| 1     | 1     | 1      | 0      | 2              |\n",
+    "\n",
+    "The circuit that we will code to implement a half adder (with an additional unimportant phase) is as follows\n",
```

Comment:
Here we should be clearer about the additional phase because it is now becoming a proper example. We should add some sentence like. "Please note that for simplicity we implement a half adder that picks up an additional phase. This circuit only works as a half-adder if the result is measured immediately. A half-adder implementation without an extra phase would require a decomposition of the Toffoli gate which requires more than three CNOT gates."